### PR TITLE
format less with prettier

### PR DESCRIPTION
**Pull request by MaxBittker**
this upgrades prettier, enables formatting for less files in the commit hook, and runs a bulk format over our existing .less 

it also updates a bunch of eslint deps to make things compatible. The resulting bulk js format is included in this PR

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by MaxBittker** [comment]
i'm gonna follow up with a jsx trailing comma PR

File: src/sentry/lint/engine.py [hunk 1]
```diff
@@ -110,7 +117,7 @@ def js_lint(file_list=None):
     return has_errors
 
 
-PRETTIER_VERSION = "1.2.2"
+PRETTIER_VERSION = "1.5.3"
```
Comment (benvinegar) [inline src/sentry/lint/engine.py]: Upgrading will almost certainly re-write a whole ton of JavaScript. You'll need to re-run prettier on JS too (and in getsentry).

**Comment by MaxBittker** [comment]
holding off on this until prettier 1.6 drops

**Comment by billyvg** [comment]
prettier issue to track https://github.com/prettier/prettier/issues/2589

**Comment by ckj** [comment]
@MaxBittker I'd like to do this, but it might be worth submitting a new PR since there would be so many conflicts at this point.

**Comment by MaxBittker** [comment]
closing in favor of #6459

**Pull request closed without merging**